ReText/editor.py

```diff
@@ -181,13 +181,28 @@ def contextMenuEvent(self, event):
 		actions = [self.parent.act(sug, trig=self.fixWord(sug)) for sug in suggestions]
 		menu = self.createStandardContextMenu()
 		menu.insertSeparator(menu.actions()[0])
+		menu.insertAction(menu.actions()[0], self.parent.act(self.tr('Add to dictionary'), trig=self.learnWord(word)))
+		menu.insertSeparator(menu.actions()[0])
 		for action in actions[::-1]:
 			menu.insertAction(menu.actions()[0], action)
 		menu.exec(event.globalPos())
 
 	def fixWord(self, correctword):
 		return lambda: self.insertPlainText(correctword)
 
+	def learnWord(self, newword):
+		return lambda: self.addNewWord(newword)
+
+	def addNewWord(self, newword):
+		cursor = self.textCursor()
+		cursor.clearSelection()
+		self.setTextCursor(cursor)
+		dictionary = self.tab.highlighter.dictionary
+		if (dictionary is None) or not newword:
+			return
+		dictionary.add(newword)
+		self.tab.highlighter.rehighlight()
```

Comment:
Rehighlighting the whole document is slow. Maybe you can use [rehighlightBlock](http://doc.qt.io/qt-5/qsyntaxhighlighter.html#rehighlightBlock)?

Reply:
Done